pkg/cmd/validate.go

```diff
@@ -97,12 +97,12 @@ func (c *commandFlags) Run(cmd *cobra.Command, args []string) error {
 	factory, err := validatorfactory.New(
 		openapiclient.NewOverlay(
 			// apply user defined patches on top of the final schema
-			openapiclient.PatchLoaderFromDirectory(filepath.Base(c.schemaPatchesDir), os.DirFS(filepath.Dir(c.schemaPatchesDir))),
+			openapiclient.PatchLoaderFromDirectory(os.DirFS(filepath.Dir(c.schemaPatchesDir)), filepath.Base(c.schemaPatchesDir)),
 			openapiclient.NewComposite(
 				// consult local OpenAPI
-				openapiclient.NewLocalFiles(c.localSchemasDir),
+				openapiclient.NewLocalFiles(nil, c.localSchemasDir),
```

Comment:
It was my initial idea but error messages will look ugly

Reply:
I would rather allow `nil` and use the helpers in the PatchLoader ;-)

Reply:
> It was my initial idea but error messages will look ugly

What happens to the errors? Is that the only drawback?

Adding something like this stanza would allow us to remove nil-checking and helper utils, while still allowing the user to provide `nil`: a strict benefit imo, unless the errors or something else is made egregious by doing that.
```
if fs == nil {
     fs = os.DirFS(filepath.Dir(dir))
     dir = filepath.Base(dir)
}
```

Reply:
This kind of error, with an `fs` the reported path is not consistent with what the user provided (could be quite confusing from an end user perspective):

```golang
	files, err := readDir(k.fs, k.dir)
	if err != nil {
		return nil, fmt.Errorf("error listing %s: %w", k.dir, err)
	}
```

Reply:
Does `os.DirFS("/")` work?

Reply:
`os.DirFS("/")` dunno, sounds uggly

Reply:
Fair enough, thanks for entertaining this. In this case yeah please do as you suggested and change patch loader for consistency

Reply:
Done